src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/GeneratedDocumentSynchronizer.cs

```diff
@@ -28,6 +28,13 @@ public void DocumentProcessed(RazorCodeDocument codeDocument, IDocumentSnapshot
             return;
         }
 
+        // If the document has been removed from the project, then don't do anything, or version numbers will be thrown off
+        if (_projectManager.GetLoadedProject(document.Project.Key) is not { } project ||
+            !project.ContainsDocument(document.FilePath))
+        {
```

Comment:
Yes, thank you, that was silly of me. Adding "Rename `GetLoadedProject` to `GetRequiredProject`" to my list. Shows how long its been since I've worked in this area :)